src/c++11/impl.cc

```diff
@@ -230,7 +231,11 @@ void SpanImpl::FinishSpan(SpanFinishOptions opts) {
   auto tags = span.mutable_tags();
 
   if (ref_.valid()) {
-    *tags->Add() = util::make_kv(ParentSpanGUIDKey, ref_.referenced().span_id());
+    // Note: Converting span_id to a hex string here is necessary
+    // because the Proto->Thrift conversion does not know to format
+    // int-valued tags as hex-valued, yet the string value for
+    // parent_span_guid is interpreted as a hex value downstream.
+    *tags->Add() = util::make_kv(ParentSpanGUIDKey, uint64ToHex(ref_.referenced().span_id()));
```

Comment:
@jmacd I realized that the thing I wanted already exists, is supported by the collectors, and would have prevented this bug in the first place: https://github.com/lightstep/lightstep-tracer-common/blob/61d9d0cb129b396931640ec9c9769e06730de2df/collector.proto#L47

Let's just use the `references` field directly.